Adding Ability to Query Table By ID

davidov541 opened the pull request:
Added the ability to query for the desired table by ID, instead of relying on ordering. 

Also added automated tests for current functionality and added functionality.

davidov541 committed:
Adding initial test for a single table.

davidov541 committed:
Adding test to make sure we handle multiple tables correctly.

davidov541 committed:
Adding test for conversion of character references.

davidov541 committed:
Adding test for not decoding the character references.

davidov541 committed:
Adding test for handling internal tags for cells.

davidov541 committed:
Adding capability to query table by name instead of relying on ordering.

schmijos commented on html_table_parser/parser.py:
```diff
@@ -33,11 +33,17 @@ def __init__(
         self._current_cell = []
         self.tables = []
+        self.named_tables = {}
+        self.current_table_name = ""
```
Since `name` is also a valid HTML attribute, I'd change naming here to `current_table_id`.

schmijos reviewed (commented):
Very nice! And thank you very much for the tests!

Two suggestions for improvement:
* Rename `current_table_name` and `name` to `current_table_id` and `id`
* Add a test for nested tables (we should be able to handle this)

One question:
* You checked-in your `.vscode` project configuration. That's not a best practice in Python right?
  @ypid What do you say?

ypid commented:
> You checked-in your .vscode project configuration. That's not a best practice in Python right?

I agree. The `.vscode` looks very generic. Is there anything special in their? I would not add editor specific dotfiles to the repo. They also don’t need to go into the `.gitignore` file of the project. If your editor creates such files/directories, you should ignore them on your local machine. If there is really a need for editor config, check out https://editorconfig.org/.

davidov541 committed:
Adding vscode folder to gitignore.

davidov541 committed:
Renaming variable to match suggestion.

davidov541 committed:
Adding test for nested table use case.

davidov541 committed:
Removing .vscode files.

davidov541 committed:
Updating README with information on how to run the tests.

davidov541 committed:
Fixing name of test_input so it no longer collides.

davidov541 commented:
@ypid @schmijos : I made the requested changes. 

I apologize for the mistake about the .vscode, that was a silly mistake on my end. I've removed the files and added the folder to .gitignore so it doesn't happen to future contributors.

I also updated the README to have instructions on how to run the tests.

Let me know if there is anything else you'd like to see before pushing this!

schmijos commented:
@davidov541 I finally took the time to look properly at your changes. Thank you very much! That you apparently like Münchner Dunkel (Lager or Bock?), actually strengthened my trust in you.

I made you collaborator. Feel free to push this library as you like.

davidov541 commented:
@schmijos Thanks! I will definitely keep an eye on the repo for you and take care of anything that comes up if I have the bandwidth.

And for the record, I'm a bock guy, but in Munich a few years back I had plenty of both!
